lib/ansible/galaxy/role.py

```diff
@@ -379,7 +379,11 @@ def install(self):
                                     if n_part != '..' and not n_part.startswith('~') and '$' not in n_part:
                                         n_final_parts.append(n_part)
                                 member.name = os.path.join(*n_final_parts)
-                                role_tar_file.extract(member, to_native(self.path))
+                                # deprecated: description='extract fallback without filter' python_version='3.11'
+                                if hasattr(tarfile, 'data_filter'):
+                                    role_tar_file.extract(member, to_native(self.path), filter='data')  # type: ignore[call-arg]
+                                else:
+                                    role_tar_file.extract(member, to_native(self.path))
```

Comment:
I think it's easier to maintain if it's more linear:
```suggestion
                                maybe_filter_kwargs = {
                                    'filter': 'data',
                                } if hasattr(tarfile, 'data_filter') else {}
                                role_tar_file.extract(
                                    member, to_native(self.path),
                                    **maybe_filter_kwargs,
                                )

```

Reply:
(moreover, this var could be extracted into some reusable module and imported into all the places)

Reply:
These aren't used enough to make it worthwhile to split them into reusable code. The `extract` method is only used once. The implementation can't be shared between ansible and ansible-test, leaving only a single occurrence of `extractall` in ansible and another in an integration test.

I think the existing conditionals are easier to read, so I wouldn't change them.